### PR TITLE
glob mechanism updated from wdl4s

### PR DESCRIPTION
**Pull request by danbills**
This incorporates the changes made from Wdl4s' globbing implementation seen here:
https://github.com/broadinstitute/wdl4s/pull/225/files

Basically a one-line change in `StandardAsyncExecutionActor` and a few details.

### CODE REVIEW

**Comment by danbills** [comment]
All the dependencies I removed are pulled in transitively from wdl4s.  T

File: backend/src/main/scala/cromwell/backend/standard/StandardAsyncExecutionActor.scala [hunk 1]
```diff
@@ -182,7 +182,7 @@ trait StandardAsyncExecutionActor extends AsyncBackendJobExecutionActor with Sta
     val rcPath = cwd./(jobPaths.returnCodeFilename)
     val rcTmpPath = rcPath.plusExt("tmp")
 
-    val globFiles = backendEngineFunctions.findGlobOutputs(call, jobDescriptor)
+    val globFiles: Set[WdlGlobFile] = call.globFiles(jobDescriptor.fullyQualifiedInputs)
```
Comment (Horneth) [inline backend/src/main/scala/cromwell/backend/standard/StandardAsyncExecutionActor.scala]: ToL: I think I lied and that we *can* (doesn't mean we have to) do this with what we have already:

```scala
def findGlobOutputs(call: TaskCallNode, jobDescriptor: BackendJobDescriptor): ErrorOr[List[WdlGlobFile]] = {
    call.callable.outputs.flatTraverse[ErrorOr, WdlGlobFile] {
      _.expression.evaluateFiles(jobDescriptor.fullyQualifiedInputs, this, WdlAnyType) map {
        _.toList collect { case glob: WdlGlobFile => glob }
      }
    }
  }
```

Comment (danbills) [reply backend/src/main/scala/cromwell/backend/standard/StandardAsyncExecutionActor.scala]: By all means, let's use what we already have!  Does this fall in `TaskDefinition`?  I.e. what is `this`?

Comment (Horneth) [reply backend/src/main/scala/cromwell/backend/standard/StandardAsyncExecutionActor.scala]: It falls in the `GlobFunctions` trait. `this` is the `IoFunctionSet`. WDL needs it (or at least claims it does...) to find out what the globs are.

**Comment by danbills** [comment]
using approach suggested by @Horneth in #2656

**Pull request closed without merging**